lib/prompts.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+var _ = require("lodash");
+var async = require("async");
+var inquirer = require("inquirer");
+
+/**
+ * Prompt user for input, validate, and add derived fields to final data object.
+ *
+ * @param {Object}    init      Initialization configuration (`prompts`, `derived`)
+ * @param {Function}  callback  Calls back with `(err, data)`
+ * @returns {void}
+ */
+module.exports = function (init, callback) {
+  if (!init) { return callback(new Error("Invalid init object")); }
+
+  // Mutate `prompts` into appropriate format.
+  var prompts = init.prompts || [];
+  if (!_.isArray(prompts)) {
+    if (!_.isObject(prompts)) {
```

Comment:
I'm not following. My logic is:

1. If an array, do nothing.
2. If not an array:
2.a. If not an object throw an error 
2.b. Otherwise, convert object to an array.

Do you have a better way to decompose this logic?